File: src/main/java/com/pedigree/render/RenderHighlightState.java
package com.pedigree.render;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

/**
 * Holds the current set of selected ids for rendering highlight.
 * This is a simple global bridge between selection sources and renderer.
 */
public final class RenderHighlightState {
    private static volatile Set<String> selectedIds = Collections.emptySet();

    private RenderHighlightState() { }

    public static void setSelectedIds(Set<String> ids) {
        if (ids == null || ids.isEmpty()) {
            selectedIds = Collections.emptySet();
        } else {
            selectedIds = Collections.unmodifiableSet(new HashSet<>(ids));
        }
    }

    public static Set<String> getSelectedIds() {
        return selectedIds;
    }
}

File: src/main/kotlin/com/family/tree/render/RenderHighlightState.kt
package com.family.tree.render

import java.util.*
import java.util.Collections.unmodifiableSet

/**
 * Holds the current set of selected ids for rendering highlight.
 * This is a simple global bridge between selection sources and renderer.
 */
object RenderHighlightState {
    @Volatile
    private var selectedIds: Set<String> = emptySet()

    @JvmStatic
    fun setSelectedIds(ids: Set<String>?) {
        selectedIds = if (ids == null || ids.isEmpty()) {
            emptySet()
        } else {
            unmodifiableSet(HashSet(ids))
        }
    }

    @JvmStatic
    fun getSelectedIds(): Set<String> = selectedIds
}
